server/src/main/java/org/elasticsearch/cluster/coordination/FollowersChecker.java

```diff
@@ -327,16 +327,16 @@ public void handleException(TransportException exp) {
                         failureCountSinceLastSuccess++;
 
                         final String reason;
-                        if (failureCountSinceLastSuccess >= followerCheckRetryCount) {
```

Comment:
We should check the failure count last so we can report more specific errors (`disconnected` etc) if available.